manifests/setup.pp

```diff
@@ -1,49 +1,36 @@
 # Handles initialization and setup of the Rails app
 class katello_devel::setup {
 
-  if $katello_devel::use_rvm {
+  if $::katello_devel::use_rvm or $::katello_devel::manage_bundler {
 
-    $seed_env = {
-      'SEED_ORGANIZATION'   => $::katello_devel::initial_organization,
-      'SEED_LOCATION'       => $::katello_devel::initial_location,
-      'SEED_ADMIN_PASSWORD' => $::katello_devel::admin_password,
-    }
+    $pidfile = "${::katello_devel::foreman_dir}/tmp/pids/server.pid"
 
-    class { '::katello_devel::rvm': } ->
-    katello_devel::rvm_bundle { 'install --without mysql:mysql2 --retry 3': } ->
-    katello_devel::rvm_bundle { 'exec rake db:migrate': } ->
-    katello_devel::rvm_bundle { 'exec rake db:seed':
-      environment => $seed_env,
-    } ~>
-    katello_devel::rvm_bundle {
-      "exec rake -- config --key ssl_ca_file --value ${katello_devel::foreman_certs::ssl_ca_cert}":
-    } ~>
-    katello_devel::rvm_bundle {
-      "exec rake -- config --key ssl_certificate --value ${katello_devel::foreman_certs::client_cert}":
-    } ~>
-    katello_devel::rvm_bundle {
-      "exec rake -- config --key ssl_priv_key --value ${katello_devel::foreman_certs::client_key}":
-    } ~>
-    exec { 'rails server':
-      cwd       => "${katello_devel::deployment_dir}/foreman",
-      command   => "sudo su ${katello_devel::user} -c '/bin/bash --login -c \"rvm use ${katello_devel::rvm_ruby} && bundle exec rails s -d\"'",
+    $seed_env = [
+      "SEED_ORGANIZATION=${::katello_devel::initial_organization}",
+      "SEED_LOCATION=${::katello_devel::initial_location}",
+      "SEED_ADMIN_PASSWORD=${::katello_devel::admin_password}",
+    ]
+
+    katello_devel::bundle { 'install --without mysql:mysql2 --retry 3 --jobs 3': } ->
+    exec { '/usr/bin/env npm install':
+      cwd       => $::katello_devel::foreman_dir,
       user      => $::katello_devel::user,
       logoutput => 'on_failure',
-      timeout   => '600',
-      path      => '/usr/local/rvm/bin:/usr/bin:/bin:/usr/bin/env',
-      unless    =>  'ps -p `cat tmp/pids/server.pid`',
-      before    => Class['foreman_proxy::register'],
-    }
-
+    } ->
+    katello_devel::bundle { 'exec rake webpack:compile': } ->
```

Comment:
Should we run this by default? There are two workflows that I know of, this one has consequences that developers may not be aware of if they are making UI changes.

Reply:
I'd prefer an option to pre-compile or use the server.  The docs show setting up a webpack server, I think that should be the default so our dev environment is most useful for all consumers.  

That's a little more complicated, we'll need some puppet-foreman support (see https://github.com/theforeman/puppet-foreman/pull/626), and creating an environment file that sets things on the right ports.

Reply:
It was a quick fix to get it running but I think we should switch to the dev server by default since that allows javascript development as well.